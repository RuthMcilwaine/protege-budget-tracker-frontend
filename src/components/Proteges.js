// Renders a list of all proteges with details, using props passed down through Routes from App.js

import React from 'react';
import { Link } from "react-router-dom";

const expenditure = (items) => items.reduce((total, item) => total + item.expenditure, 0)

const renderProteges = (protegeData) => {
  return protegeData.map((protege, index) => {
    return (
      <div className="protegelist" key={index}>
        <h2>{protege.protege.protegename}</h2>
        <ul>
          <li>Email: {protege.protege.protegeemail}</li>
          <li>Expenditure: {expenditure(protege.items)}</li>
          <li>Balance: {1000 - expenditure(protege.items)}</li>
          <li><Link to={"/proteges/"+protege.protege._id}>View/edit</Link></li>
        </ul>
      </div>
    )
  })
}

const Proteges = (props) => {
  const protegeData = props.proteges
  const displayProteges = renderProteges(protegeData)
  return displayProteges
}

export default Proteges;