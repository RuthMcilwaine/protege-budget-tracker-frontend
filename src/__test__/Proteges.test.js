import React from 'react';
import ReactDOM from 'react-dom';
import Proteges from '../components/Proteges';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

const proteges = [{
  "protege": {
  "_id": "5d26b687a7946c6d37dcfdde",
  "protegename": "El",
  "protegeemail": "Elizabeth@myob.com.au",
  "expenditure": 0,
  "balance": 1000,
  "createdAt": "2019-07-11T04:09:43.804Z",
  "updatedAt": "2019-07-18T00:37:55.952Z",
  "__v": 0
  },
  "items": [
  {
  "_id": "5d2fc1746574fd0573e88cec",
  "protege_id": "5d26b687a7946c6d37dcfdde",
  "description": "Coffee",
  "expenditure": 50,
  "date": "2019-07-18T00:46:38.730Z",
  "createdAt": "2019-07-18T00:46:44.984Z",
  "updatedAt": "2019-07-18T00:46:44.984Z",
  "__v": 0
  }, {
    "_id": "5d2fc1746574fd0573e88cee",
    "protege_id": "5d26b687a7946c6d37dcfdde",
    "description": "Coffee",
    "expenditure": 100,
    "date": "2019-07-18T00:46:38.730Z",
    "createdAt": "2019-07-18T00:46:44.984Z",
    "updatedAt": "2019-07-18T00:46:44.984Z",
    "__v": 0
    }
  ]}
]

describe('<Proteges />', () => {
  it('renders <Proteges /> components', () => {
    const wrapper = shallow(<Proteges proteges={proteges}/>);
  });
  expect(proteges).toMatchSnapshot();
})

