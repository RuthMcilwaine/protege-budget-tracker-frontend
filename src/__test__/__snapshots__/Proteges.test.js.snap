// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  Object {
    "items": Array [
      Object {
        "__v": 0,
        "_id": "5d2fc1746574fd0573e88cec",
        "createdAt": "2019-07-18T00:46:44.984Z",
        "date": "2019-07-18T00:46:38.730Z",
        "description": "Coffee",
        "expenditure": 50,
        "protege_id": "5d26b687a7946c6d37dcfdde",
        "updatedAt": "2019-07-18T00:46:44.984Z",
      },
      Object {
        "__v": 0,
        "_id": "5d2fc1746574fd0573e88cee",
        "createdAt": "2019-07-18T00:46:44.984Z",
        "date": "2019-07-18T00:46:38.730Z",
        "description": "Coffee",
        "expenditure": 100,
        "protege_id": "5d26b687a7946c6d37dcfdde",
        "updatedAt": "2019-07-18T00:46:44.984Z",
      },
    ],
    "protege": Object {
      "__v": 0,
      "_id": "5d26b687a7946c6d37dcfdde",
      "balance": 1000,
      "createdAt": "2019-07-11T04:09:43.804Z",
      "expenditure": 0,
      "protegeemail": "Elizabeth@myob.com.au",
      "protegename": "El",
      "updatedAt": "2019-07-18T00:37:55.952Z",
    },
  },
]
`;
